package Gather;

enum GtUserMessageType {
    GT_USER_LOGIN_REQUEST = 1;
    GT_USER_LOGIN_RESPONSE = 2;
    GT_USER_LOGOUT_RESPONSE = 3;
}

message GtUserLoginRequest {
    optional string user = 1;
    optional string passwd = 2;
}

message GtUserLoginResponse {
    optional int32 result = 1;
    optional string secret = 2;
}

message GtUserLogoutResponse {
    optional int32 reason = 1;
}

message GtUserLinkDest {
    optional int32 type = 1;
    optional int32 page = 2;
    optional float x = 3;
    optional float y = 4;
    optional float zoom = 5;
    optional string uri = 6;
}

message GtUserBookmark {
    optional string title = 1;
    optional GtUserLinkDest dest = 2;
    repeated GtUserBookmark children = 3;
}

message GtUserBookmarks {
    optional string id = 1;
    optional GtUserBookmark root = 2;
}

message GtUserDocRange {
    optional int32 type = 1;
    optional int32 begin_page = 2;
    optional float begin_x = 3;
    optional float begin_y = 4;
    optional int32 end_page = 5;
    optional float end_x = 6;
    optional float end_y = 7;
}

message GtUserDocNote {
    optional int32 type = 1;
    optional GtUserDocRange range = 2;
}

message GtUserDocNotes {
    optional string id = 1;
    repeated GtUserDocNote notes = 2;
}